classes/MusicHandler.js

```diff
@@ -8,12 +8,51 @@ module.exports = class MusicHandler {
 		this.player = player;
 	}
 
-	disconnect() {
+	async disconnect() {
 		const { bot } = require('../main.js');
+		const voiceChannel = bot.guilds.cache.get(this.player.guildId).channels.cache.get(this.player.channelId);
+		const { oldVoiceChannel } = require('../events/voiceStateUpdate.js');
 		clearTimeout(this.player.timeout);
 		clearTimeout(this.player.pauseTimeout);
 		this.player.disconnect();
 		bot.music.destroyPlayer(this.player.guildId);
+		if (voiceChannel?.type === 'GUILD_STAGE_VOICE') {
+			const voicePerms = bot.guilds.cache.get(this.player.guildId).channels.cache.get(this.player.channelId).permissionsFor(bot.user.id);
+			if (!voicePerms.has(['VIEW_CHANNEL', 'CONNECT', 'SPEAK'])) {
+				return;
+			}
+			if (!voicePerms.has(Permissions.STAGE_MODERATOR)) {
+				return;
+			}
+			if (voiceChannel.stageInstance?.topic === getLocale(guildData.get(`${this.player.guildId}.locale`) ?? defaultLocale, 'MUSIC_STAGE_TOPIC')) {
+				try {
+					await voiceChannel.stageInstance.delete();
+				}
+				catch (err) {
+					logger.error({ message: `${err.message}\n${err.stack}`, label: 'Quaver' });
+				}
+			}
+			return;
+		}
+		if (oldVoiceChannel?.type === 'GUILD_STAGE_VOICE') {
```

Comment:
`oldVoiceChannel?.type === 'GUILD_STATE_VOICE` would always return false because `oldVoiceChannel` is `undefined`

Reply:
Reverted to the very first way of implementation.